superset/db_engine_specs/singlestore.py

```diff
@@ -0,0 +1,558 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import logging
+import re
+from datetime import datetime
+from typing import Any, Optional
+from urllib import parse
+
+from flask import current_app
+from sqlalchemy import types
+from sqlalchemy.engine import URL
+
+from superset.constants import TimeGrain
+from superset.db_engine_specs.base import BaseEngineSpec, ColumnTypeMapping, LimitMethod
+from superset.models.core import Database
+from superset.models.sql_lab import Query
+from superset.utils.core import GenericDataType
+
+logger = logging.getLogger(__name__)
+
+
+class SingleStoreSpec(BaseEngineSpec):
```

Comment:
If you also inherit from `superset.db_engine_specs.base.BasicParametersMixin` you'll get a nicer configuration UI when adding/editing the database.

Reply:
Thanks for the suggestion. Added this.